conn/getdomain.c

```diff
@@ -99,8 +99,9 @@ int getdnsdomainname(struct Buffer *domain)
     mutt_buffer_strcpy(domain, ++p);
     rc = 0;
     mutt_debug(LL_DEBUG1, "Hostname: %s\n", mutt_buffer_string(domain));
-    freeaddrinfo(h);
   }
+  if (h)
+    freeaddrinfo(h);
```

Comment:
That'll do the trick, but I think I might have gone for:
```c
  if (h)
  {
    if (h->ai_canonname && (p = strchr(h->ai_canonname, '.')))
    {
      mutt_buffer_strcpy(domain, ++p);
      rc = 0;
      mutt_debug(LL_DEBUG1, "Hostname: %s\n", mutt_buffer_string(domain));
    }
    freeaddrinfo(h);
  }
```

Reply:
The check on `h->ai_canonname` is useless, because of the `AI_CANONNAME` flag we're passing :)

Reply:
done